src/Reflection/ReflectionClass.php

```diff
@@ -1019,16 +1030,44 @@ private function getTraitPrecedences(): array
         return $this->cachedTraitPrecedences;
     }
 
+    /**
+     * Return a list of the used modifiers when importing traits for this class.
+     * The returned array is in key/value pair in this format:.
+     *
+     *   'methodName' => 'modifier'
+     *
+     * @return array<string, int>
+     *
+     * @example
+     * // When reflecting a class such as:
+     * class Foo
+     * {
+     *     use MyTrait {
+     *         myTraitMethod as public;
+     *     }
+     * }
+     * // This method would return
+     * //   ['myTraitMethod' => 1]
+     */
+    private function getTraitModifiers(): array
+    {
+        $this->parseTraitUsages();
+
+        /** @return array<string, int> */
+        return $this->cachedTraitModifiers;
+    }
+
     private function parseTraitUsages(): void
     {
-        if ($this->cachedTraitAliases !== null && $this->cachedTraitPrecedences !== null) {
+        if ($this->cachedTraitAliases !== null && $this->cachedTraitModifiers !== null && $this->cachedTraitPrecedences !== null) {
```

Comment:
Having 3 different caches for the same thing is quite worrisome :D

Reply:
I will remove the new one and combine with one of the previous. It should be easy.

Reply:
Hmm, `getTraitAliases()` is `public` so it would be more difficult...

Reply:
I suggest to ignore it for now... :)

Reply:
Let's leave it alone - wasn't suggesting action is needed, but rather warning about ever-raising complexity.